tests/cli/commands/test_info.py

```diff
@@ -0,0 +1,177 @@
+                "postgresql+psycopg2://:PASSWORD@postgres/airflow",
+            ),
+            (
+                "postgresql+psycopg2://postgres/airflow",
+                "postgresql+psycopg2://postgres/airflow",
+            ),
+        ]
+    )
+    def test_should_remove_pii_from_url(self, before, after):
+        assert after == self.instance.process_url(before)
+
+
+class TestAirflowInfo:
+    @classmethod
+    def teardown_class(cls) -> None:
+        for handler_ref in logging._handlerList[:]:  # type: ignore
+            logging._removeHandlerRef(handler_ref)  # type: ignore
+        importlib.reload(airflow_local_settings)
+        configure_logging()
+
+    @staticmethod
+    def unique_items(items):
+        return {i[0] for i in items}
+
+    @conf_vars(
+        {
+            ("core", "executor"): "TEST_EXECUTOR",
+            ("core", "dags_folder"): "TEST_DAGS_FOLDER",
+            ("core", "plugins_folder"): "TEST_PLUGINS_FOLDER",
+            ("logging", "base_log_folder"): "TEST_LOG_FOLDER",
+            ('core', 'sql_alchemy_conn'): 'postgresql+psycopg2://postgres:airflow@postgres/airflow',
+            ('logging', 'remote_logging'): 'True',
+            ('logging', 'remote_base_log_folder'): 's3://logs-name',
+        }
+    )
+    def test_airflow_info(self):
+        importlib.reload(airflow_local_settings)
+        configure_logging()
+        instance = info.AirflowInfo(info.NullAnonymizer())
+        expected = {
+            'executor',
+            'version',
+            'task_logging_handler',
+            'plugins_folder',
+            'base_log_folder',
+            'remote_base_log_folder',
+            'dags_folder',
+            'sql_alchemy_conn',
+        }
+        assert self.unique_items(instance._airflow_info) == expected
+
+    def test_system_info(self):
+        instance = info.AirflowInfo(info.NullAnonymizer())
+        expected = {'uname', 'architecture', 'OS', 'python_location', 'locale', 'python_version'}
+        assert self.unique_items(instance._system_info) == expected
+
+    def test_paths_info(self):
+        instance = info.AirflowInfo(info.NullAnonymizer())
+        expected = {'airflow_on_path', 'airflow_home', 'system_path', 'python_path'}
+        assert self.unique_items(instance._paths_info) == expected
+
+    def test_tools_info(self):
+        instance = info.AirflowInfo(info.NullAnonymizer())
+        expected = {
+            'cloud_sql_proxy',
+            'gcloud',
+            'git',
+            'kubectl',
+            'mysql',
+            'psql',
+            'sqlite3',
+            'ssh',
+        }
+        assert self.unique_items(instance._tools_info) == expected
+
+    @conf_vars(
+        {
+            ('core', 'sql_alchemy_conn'): 'postgresql+psycopg2://postgres:airflow@postgres/airflow',
+        }
+    )
+    def test_show_info(self):
+        runner = CliRunner()
+        result = runner.invoke(info.show_info)
```

Comment:
I updated the command invocation and `stdout` retrieval (as shown here) in a few test cases. Otherwise this file is a copy of `test_info_command.py`.